src/main/java/step3/domain/Cars.java

```diff
@@ -0,0 +1,26 @@
+package step3.domain;
+
+import step3.domain.rule.MoveRule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cars {
+    private List<Car> cars = new ArrayList<>();
+
+    public Cars(int carCount) {
+        for(int i = 0; i < carCount; i++) {
+            cars.add(new Car());
+        }
+    }
+
+    public void moveAll(int roundTime, MoveRule moveRule) {
+        for(Car car : cars) {
+            car.move(roundTime, moveRule.canMove());
+        }
+    }
+
+    public List<Car> getValue() {
+        return cars;
```

Comment:
`List<Car>`의 참조를 직접적으로 노출하고 있습니다.

외부에서 add를 하거나 수정하는 행위가 가능해 보여요 아래와 같이 불변하게 반환해주시면 어떨까요?

```suggestion
        return Collections.unmodifiablelist(cars);
```

Reply:
참조타입에 대한 불변 객체 처리를 하는 방법에 대해 공부하게되었습니다. 감사합니다.